File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/outofband/OutOfBand.java
package com.evernym.verity.sdk.protocols.outofband;

import com.evernym.verity.sdk.protocols.outofband.v1_0.OutOfBandV1_0;

/**
 * Builds and sends a new encrypted agent message for the Connections protocol.
 */
public class OutOfBand {
    private OutOfBand() {
    }

    /**
     * used by invitee to send Reuse message
     *
     * @param forRelationship the relationship identifier (DID) for the pairwise
     *                        relationship that will be reused
     * @param inviteUrl the Out-of-Band invitation url
     */
    public static OutOfBandV1_0 v1_0(String forRelationship, String inviteUrl) {
        return new OutOfBandImplV1_0(forRelationship, inviteUrl);
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/OutOfBand/OutOfBand.cs
using System.Json;

namespace VeritySDK.Protocols.OutOfBand
{
    /// <summary>
    /// The OutOfBand protocol allow an interaction intent, connection information and reuse of a relationship.
    /// </summary>
    public class OutOfBand
    {
        private OutOfBand() { }

        /// <summary>
        /// used by invitee to send Reuse message 
        /// </summary>
        /// <param name="forRelationship">the relationship identifier (DID) for the pairwise relationship that will be reused</param>
        /// <param name="inviteUrl">the Out-of-Band invitation url</param>
        /// <returns></returns>
        public static OutOfBandV1_0 v1_0(string forRelationship, string inviteUrl)
        {
            return new OutOfBandV1_0(forRelationship, inviteUrl);
        }
    }
}